regimes/gr/README.md

```diff
@@ -0,0 +1,117 @@
+# 🇬🇷 GOBL Greece Tax Regime
+
+Greece uses the myDATA and Peppol BIS Billing 3.0 formats for their e-invoicing/tax-reporting system.
+
+Example GR GOBL files can be found in the [`examples`](./examples) (YAML uncalculated documents) and [`examples/out`](./examples/out) (JSON calculated envelopes) subdirectories.
+
+## Public Documentation
+
+* [myDATA API Documentation v1.0.7](https://www.aade.gr/sites/default/files/2023-10/myDATA%20API%20Documentation_v1.0.7_eng.pdf)
+* [Greek Peppol BIS Billing 3.0](https://www.gsis.gr/sites/default/files/eInvoice/Instructions%20to%20B2G%20Suppliers%20and%20certified%20PEPPOL%20Providers%20for%20the%20Greek%20PEPPOL%20BIS-EN-%20v1.0.pdf)
+* [VAT Rates](https://www.gov.gr/en/sdg/taxes/vat/general/basic-vat-rates)
+
+## Greece specifics
+
+### Islands' reduced tax rates
+
+Greece has three VAT rates: standard, reduced and super-reduced. These rates are reduced by a 30% on the islands of Leros, Lesbos, Kos, Samos and Chios.
+
+GOBL invoices including the `islands` tag inside the invoice's tax section will get the reduced rates applied instead of the regular ones. For example:
+
+```js
+{
+  "$schema": "https://gobl.org/draft-0/bill/invoice",
+  //...
+  "tax": {
+    "tags": [
+      "islands"
+    ]
+  },
+  // ...
+  "lines": [
+    {
+      "i": 1,
+      "quantity": "20",
+      "item": {
+        "name": "Υπηρεσίες Ανάπτυξης",
+        "price": "90.00",
+      },
+      "sum": "1800.00",
+      "taxes": [
+        {
+          "cat": "VAT",
+          "rate": "standard",
+          "percent": "17%" // Reduced island rate applied instead of the regular 24%
+        }
+      ],
+      "total": "1800.00"
+    }
+  ],
+}
+```
```

Comment:
I'm not 100% convinces about the approach for the alternative rates for some Greek islands. For two reasons:

1. In other countries we use a different approach for a similar problem: in [PT we use a tag extension](https://github.com/invopop/gobl/blob/main/regimes/pt/tax_categories.go#L46-L48) and [in MX we use a separated rate](https://github.com/invopop/gobl/blob/main/regimes/mx/tax_categories.go#L54-L55) for the border regions.
2. This solution forces us to [split the VAT Category Code](https://github.com/invopop/gobl/pull/307/files#diff-b511816f950af7d992d1ecabd067109dda861e5861765c0d5dd01dd417f609afR51-R52) that we'll need to provide to the IAPR.

An alternative approach would be to use separated rates but using combined keys [like we do in Spain for tax rate equivalence](https://github.com/invopop/gobl/blob/main/regimes/es/tax_categories.go#L82):
* `standard`
* `reduced`
* `super-reduced`
* `standard+islands`
* `reduced+islands`
* `super-reduced+islands`

This solution would let us avoid the IAPR vat category split mentioned in point 2 above.

BTW, `islands` is the best code/tag I've been able to come up with, but I'm open to suggestions here too.

Reply:
Hehe nice. So the key question is, ¿do we need to identify the different rates like in Portugal? If not, then I'd be tempted to not worry about this and just let the client put whatever rates they like in for the exception cases.

If rate keys or any other data are required, then I'd consider going with extensions with a code for the region that has a reduced rate.

As you say, `islands` is too generic and doesn't really give enough information about where. Nor does it deal with future exception cases.

Reply:
So I see there is indeed a code for the VAT category with 6 options in total. I think the best bet is to add this property as an extension. We want to avoid using tags at the document level as they don't account for mixing rates in the same document and don't coincide with how most ERPs deal with tax on a per-line basis.

We may need to write some extra code to handle these cases though so that the extension is added automatically when a rate is provided, and validated to ensure it doesn't conflict... I'll add some more comments in-line.